astropy/io/fits/hdu/compressed.py

```diff
@@ -1417,7 +1417,7 @@ def data(self, data):
     def compressed_data(self):
         # First we will get the table data (the compressed
         # data) from the file, if there is any.
-        compressed_data = super(BinTableHDU, self).data
+        compressed_data = super().data
```

Comment:
I think this can go wrong only when someone subclasses `class Mine(CompImageHDU, MySpecial)` - and they don't override data in `Mine`, but do in `MySpecial` - with this change `MySpecial.data` will get called while before it would not. I find it hard to see how this is not an improvement, though! (And doubt these classes are subclassed much anyway.)

Reply:
Yes, I believe it was just a copy&paste mistake that worked correctly. However as with almost all these non-trivial `super` arguments: You can't be really sure. Except for the other case which went into an recursionerror if one subclasses `_ErrList`.

Reply:
Agreed about this one.